rest/route/github.go

```diff
@@ -315,20 +304,23 @@ func (gh *githubHookApi) handleCheckRunRerequested(ctx context.Context, event *g
 			"task":    checkRunTask,
 			"message": "finding task",
 		})
-		return gimlet.NewJSONInternalErrorResponse(errors.Wrapf(taskErr, "finding task '%s' for check run", checkRunTask))
+		catcher.Add(errors.Wrapf(taskErr, "finding task '%s' for check run", checkRunTask))
+	}
+	if !utility.StringSliceContains(evergreen.TaskCompletedStatuses, taskToRestart.Status) {
+		return errors.Errorf("task '%s' is not in a completed state", checkRunTask)
 	}
-	githubUser, err := user.FindByGithubUID(int(event.GetSender().GetID()))
+	githubUser, err := user.FindByGithubUID(uid)
 	if err != nil {
 		grip.Error(message.WrapError(err, message.Fields{
 			"source":  "GitHub hook",
 			"msg_id":  gh.msgID,
 			"event":   gh.eventType,
 			"owner":   owner,
 			"repo":    repo,
-			"user":    event.GetSender().GetID(),
+			"user":    uid,
 			"message": "finding user by GitHub ID",
 		}))
-		return gimlet.NewJSONInternalErrorResponse(errors.Wrapf(err, "finding user by GitHub ID '%d'", event.GetSender().GetID()))
+		catcher.Add(errors.Wrapf(err, "finding user by GitHub ID '%d'", uid))
 	}
 	if err := model.ResetTaskOrDisplayTask(ctx, gh.settings, taskToRestart, githubUser.Id, evergreen.GithubCheckRun, false, nil); err != nil {
```

Comment:
githubUser.Id will panic if we use catcher instead of returning when the github user is nil

Reply:
will error if the user is not foud because we do not want random people on github to be able to restart our tasks